build/Dockerfile

```diff
@@ -12,6 +12,24 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+FROM registry.access.redhat.com/ubi9/python-311 as wheel
+
+ARG WHEEL_VERSION=""
+USER root
+RUN --mount=type=cache,target=/root/.cache/pip \
+    python -m pip install --upgrade pip && \
+    python -m pip install build
+COPY tuning tuning
+RUN --mount=source=.git,target=.git,type=bind \
+    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+    # build wheel if wheel version is empty else download the wheel from PyPi
+    if [[ -z "${WHEEL_VERSION}" ]]; \
+    then python -m build --wheel --outdir /tmp; \
```

Comment:
When I try to build the image using `docker build --progress=plain -t fms-hf-tuning:0.0.1 . -f build/Dockerfile` I get an error:
```
[1/2] STEP 6/6: RUN --mount=source=.git,target=.git,type=bind     --mount=type=bind,source=pyproject.toml,target=pyproject.toml     if [[ -z "${WHEEL_VERSION}" ]];     then python -m build --wheel --outdir /tmp;     else pip download fms-hf-tuning==${WHEEL_VERSION} --dest /tmp --only-binary=:all: --no-deps;     fi &&     ls /tmp/*.whl >/tmp/bdist_name
* Creating isolated environment: venv+pip...
ERROR Source /opt/app-root/src does not appear to be a Python project: no pyproject.toml or setup.py
Error: building at STEP "RUN --mount=source=.git,target=.git,type=bind --mount=type=bind,source=pyproject.toml,target=pyproject.toml if [[ -z "${WHEEL_VERSION}" ]];     then python -m build --wheel --outdir /tmp;     else pip download fms-hf-tuning==${WHEEL_VERSION} --dest /tmp --only-binary=:all: --no-deps;     fi &&     ls /tmp/*.whl >/tmp/bdist_name": while running runtime: exit status 1
```

Reply:
I suppose the error is caused by using Podman instead of Docker

Reply:
hmm the build command worked for me on a RHEL9.2 x86 VM
```
podman build --build-arg WHEEL_VERSION=0.0.1 --progress=plain -t fms-hf-tuning:0.0.1 . -f build/Dockerfile
```
I wonder podman machine failed to mount the pyproject.toml from your local host file system.  If your local build uses `podman machine` on MacOS Intel. You should mount your workspace into the podman machine first
For example:
```
podman rm
podman machine init --cpus 6 --memory 8196 -v /Users/sutaakar:/Users/sutaakar --rootful --now
```

Reply:
Just an idea, what about using:
```
COPY .git .git
COPY pyproject.toml pyproject.toml
# build wheel if wheel version is empty else download the wheel from PyPi
RUN if [[ -z "${WHEEL_VERSION}" ]]; \
    then python -m build --wheel --outdir /tmp; \
    else pip download fms-hf-tuning==${WHEEL_VERSION} --dest /tmp --only-binary=:all: --no-deps; \
    fi && \
    ls /tmp/*.whl >/tmp/bdist_name
```
This way there won't be any issue with mounting files and should work on all platforms. It won't affect result image the files are copied on builder image only.

Reply:
Great idea. Thanks.